README-mux.md

```diff
@@ -0,0 +1,345 @@
+# Setting Up ViaQ Logging
+
+## Intro
+
+ViaQ Logging is based on the [OpenShift
+Logging](https://github.com/openshift/origin-aggregated-logging) stack.  You
+can use either the OpenShift Container Platform (OCP) based on RHEL7, or
+OpenShift Origin (Origin) based on CentOS7.
+
+The component which uses the secure_forward listener in OpenShift is called
+*mux*, short for multiplex, because it acts like a multiplexor or manifold,
+taking in connections from many collectors, and distributing them to a data
+store.  Another term would be FEAN, short for
+Forward/Enrich/Aggregate/Normalize.
+
+This document uses the term *mux* to refer to this component.
+
+## Provisioning a machine to run ViaQ
+
+**WARNING** DO NOT INSTALL `libvirt` on the OpenShift machine!  You will run
+  into all sorts of problems related to name resolution and DNS.  For example,
+  your pods will not start, will be in the Error state, and will have messages
+  like this: `tcp: lookup kubernetes.default.svc.cluster.local: no such host`
+
+ViaQ on OCP requires a RHEL 7.3 or later machine.  ViaQ on Origin requires a
+up-to-date CentOS 7 machine.  You must be able to ssh into the machine using an
+ssh keypair.  This means you will need to:
+
+* provide a ssh pubkey
+* create a user account (or use root)
+* add the ssh pubkey to the user account `~/.ssh/authorized_keys`
+  * or use `ssh-copy-id -i /path/to/my/id_rsa.pub ...` to copy one over
+* enable passwordless sudo (if not using root above) e.g. in sudoers config:
+  * `centos ALL=(ALL) NOPASSWD:ALL`
+* allow connections on the following ports/protocols:
+  * icmp (for ping)
+  * tcp ports 22, 80, 443, 8443 (openshift console), 24284 (secure_forward)
+
+This will allow you to access the machine via ssh (in order to run Ansible -
+see below), to access the external services such as Kibana and mux, and to
+access the OpenShift UI console.
+
+ViaQ on OCP requires a RHEL and OCP subscription.
+
+ViaQ on Origin requires these [Yum Repos](#appendix-1-centos7-viaq-yum-repos).
+
+You will need to install the following packages: docker iptables-services NetworkManager
+
+You will need to configure sudo to not require a tty e.g. something like this:
+
+    # ls -l /etc/sudoers.d/999-cloud-init-requiretty
+    -r--r-----. 1 root root 21 Dec  1 13:17    /etc/sudoers.d/999-cloud-init-requiretty
+    # cat /etc/sudoers.d/999-cloud-init-requiretty
+    Defaults !requiretty
+
+You may need to disable SELinux `sudo setenforce Permissive` or edit
+`/etc/selinux/config` and use `SELINUX=permissive`
+
+You will need to ensure NetworkManager is running:
+
+    # systemctl daemon-reload
+    # systemctl enable NetworkManager
+    # systemctl start NetworkManager
+
+## Installing ViaQ
+
+These instructions and config files are for an all-in-one, single machine, run
+ansible on the same machine you are installing ViaQ on.
+
+*NOTE* *THIS IS NOT FOR PRODUCTION USE*
```

Comment:
Yes.  Because the default user permissions are wide open - any user can login.  And because I don't want to be responsible if someone uses these instructions to try to run something in production that breaks or causes a massive security problem.